api/envoy/service/ext_proc/v3/external_processor.proto

```diff
@@ -56,9 +56,31 @@ service ExternalProcessor {
   }
 }
 
+// This message specifies the filter protocol configurations which will be sent to the ext_proc
+// server in a :ref:`ProcessingRequest <envoy_v3_api_msg_service.ext_proc.v3.ProcessingRequest>`.
+// If the server does not support these protocol configurations, it may choose to close the gRPC stream.
+// If the server supports these protocol configurations, it should respond based on the API specifications.
+message ProtocolConfiguration {
+  // Specify the filter configuration :ref:`request_body_mode
+  // <envoy_v3_api_field_extensions.filters.http.ext_proc.v3.ProcessingMode.request_body_mode>`
+  envoy.extensions.filters.http.ext_proc.v3.ProcessingMode.BodySendMode request_body_mode = 1
+      [(validate.rules).enum = {defined_only: true}];
+
+  // Specify the filter configuration :ref:`response_body_mode
+  // <envoy_v3_api_field_extensions.filters.http.ext_proc.v3.ProcessingMode.response_body_mode>`
+  envoy.extensions.filters.http.ext_proc.v3.ProcessingMode.BodySendMode response_body_mode = 2
+      [(validate.rules).enum = {defined_only: true}];
+
+  // Specify the filter configuration :ref:`send_body_without_waiting_for_header_response
+  // <envoy_v3_api_field_extensions.filters.http.ext_proc.v3.ExternalProcessor.send_body_without_waiting_for_header_response>`
+  // If Envoy is waiting for a header response from the server, setting ``true`` means Envoy will send body to the server
```

Comment:
Please replace "Envoy" with "the client" in all of these comments.  Envoy is not the only xDS data plane. :)

Reply:
Updated for the changes related to this PR